ddcommon/src/azure_app_services.rs

```diff
@@ -93,6 +94,15 @@ impl AzureMetadata {
             .map(|v| v.to_string())
     }
 
+    fn extract_resource_group(s: Option<String>) -> Option<String> {
+        let re: Regex = Regex::new(r"(.+)\+(.+)-(.+)-(.+)").unwrap();
```

Comment:
We typically avoid using `unwrap()` as if the call fails, it'll crash our function.
Here it's probably fine as the regex is static (so it [shouldn't error](https://docs.rs/regex/latest/regex/struct.Regex.html#method.new)).

I'm not sure what the best answer here would be, maybe matching and returning a safe default?

Reply:
I added some unit tests to validate the behavior of the `extract_resource_group` regex function. If the pattern matches we return the resource group as designed. If the pattern doesn't match then `None` is returned and the resource group resolves to an unknown value gracefully.

Let me know if you think that's sufficiently safe! All of the examples in the regex docs use `unwrap()` so I wasn't sure what the best alternative was.
https://docs.rs/regex/latest/regex/struct.Regex.html#method.captures

https://github.com/DataDog/libdatadog/blob/88c6c960320842194f342d214e685762385fca65/ddcommon/src/azure_app_services.rs#L401-L448